kernel/kernel-api/src/main/java/io/delta/kernel/client/ExpressionHandler.java

```diff
@@ -34,6 +34,16 @@
  */
 @Evolving
 public interface ExpressionHandler {
+    /**
+     * Is the given expression evaluation supported on the data with given schema?
+     *
+     * @param inputSchema Schema of input data on which the expression is evaluated.
+     * @param expression Expression to evaluate.
+     * @param outputType Expected result data type.
+     * @return true if supported and false otherwise.
+     */
+    boolean isSupported(StructType inputSchema, Expression expression, DataType outputType);
```

Comment:
nit: IMO `expression` should come first, eh?

Reply:
Tried to keep it the same as the existing APIs where the pattern is: input data type -> expression -> output type.